rpc/auth/guest/provider.go

```diff
@@ -0,0 +1,119 @@
+package guest
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/0xsequence/ethkit/go-ethereum/common/hexutil"
+	ethcrypto "github.com/0xsequence/ethkit/go-ethereum/crypto"
+	"github.com/0xsequence/go-sequence/intents"
+	"github.com/0xsequence/waas-authenticator/proto"
+	"github.com/0xsequence/waas-authenticator/rpc/attestation"
+	"github.com/0xsequence/waas-authenticator/rpc/auth"
+	"github.com/0xsequence/waas-authenticator/rpc/tenant"
+)
+
+type AuthProvider struct{}
```

Comment:
Do you have any tests for those new providers?

Reply:
Added some basic success-scenario tests. Planning to expand them much further later on